### ISSUE TITLE
Scroll widget box sizing

### ISSUE THREAD

**Issue by nimble0** [mikke89/RmlUi#314]
I'm a little confused by the logic in these two places where the containing block's width is used for both the width and height.
 https://github.com/mikke89/RmlUi/blob/d56f17e49ca2ee88aadeb304228cd1eae14e3f51/Source/Core/Elements/WidgetSlider.cpp#L256
https://github.com/mikke89/RmlUi/blob/d56f17e49ca2ee88aadeb304228cd1eae14e3f51/Source/Core/WidgetScroll.cpp#L221
It's apparently intentional given the comments above the lines but I don't see how it makes sense to turn the element box into a square.

**Comment by mikke89** [comment]
Hm, I can only guess based on those comments. Normally the containing block is a definite size, but I guess they had some cases where it wasn't and had to resolve that somehow. Then I guess the width is just as good as any other arbitrary value...

How did you end up down this road, is this causing any issues or just curiosity? :)

**Comment by nimble0** [comment]
Yeah, I'm trying to center a slider track in the middle of its parent (a table cell) but it centers in the middle of this square instead. It works if I change WidgetSlider.cpp. Here's an example in the demo sandbox:

RML
```
<table>
<tr>
<td>boo</td>
<td><input type="range" /></td>
</tr>
</table>
```
RCSS
```
table
{
font-size: 5em;
}

td
{
border: 1dp #000;
}

input
{
width: 100%;
height: 100%;
}

input.range slidertrack
{
height: 6dp;
margin: auto;
background-color: #7bd6da;
border-width: 0.25em;
border-color: #0000;
}
input.range sliderbar
{
box-sizing: border-box;
width: 6dp;
height: 100%;
border-width: 0.25em 0em;
border-color: #0000;
background-color: #000;
}
```
Expected result:
![slider-expected](https://user-images.githubusercontent.com/9959776/169598194-8cf99759-6ddb-4e9d-a4fe-705481d0374b.png)
Actual result:
![slider-actual](https://user-images.githubusercontent.com/9959776/169598191-5ae93695-6404-41e4-afdf-651bffa5daa1.png)

Interestingly, the slider bar doesn't have the same problem despite its height being set to 100%.

**Comment by mikke89** [comment]
Ah, I see, seems like a useful thing. Good catch finding where the issue is.

I wouldn't mind changing it to use the containing block height. Just in case, we can check whether it is negative and if so set it to the containing block width, or perhaps setting it to zero makes more sense? That case should be extremely rare anyway, so we should be relatively safe.

**Comment by nimble0** [comment]
Setting it to 0 sounds like it would make more sense but without knowing which cases these changes were intended to fix it's hard to be sure.

**Comment by mikke89** [comment]
Okay, I've really tried to force it into the situation they intended to fix, and it seems not possible.

- For widget slider: The containing block is the parent `input` element. This is a replaced element and so always has a height.
- For widget scroll: The containing block is the element the scroller is attached to. The formatting is always called while closing the element which means that its height was just determined.

Conclusion: We can simply pass the containing block size directly. No checks needed.

Could you submit a PR? :)

**Comment by mikke89** [comment]
Fixed in #321, thanks!
